ServerCore/Startup.cs

```diff
@@ -34,9 +36,21 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddAuthentication().AddMicrosoftAccount(microsoftOptions =>
             {
-                microsoftOptions.ClientId = Configuration["Authentication:Microsoft:ApplicationId"];
-                microsoftOptions.ClientSecret = Configuration["Authentication:Microsoft:Password"];
+                microsoftOptions.ClientId = Configuration["Authentication-Microsoft-ApplicationId"];
+                microsoftOptions.ClientSecret = Configuration["Authentication-Microsoft-Password"];
             });
+
+            services.AddAuthorization(options =>
+            {
+                options.AddPolicy("IsAuthor", policy => policy.Requirements.Add(new IsAuthorInEventRequirement()));
+                options.AddPolicy("IsAdmin", policy => policy.Requirements.Add(new IsAdminInEventRequirement()));
+                options.AddPolicy("IsGlobalAdmin", policy => policy.Requirements.Add(new IsGlobalAdminRequirement()));
+            });
+
+            services.AddTransient<IAuthorizationHandler, IsAuthorInEventHandler>();
```

Comment:
Would Scoped or Singleton work? It seems like we could end up creating a ton of objects

Reply:
Singleton absolutely does not work. I'm looking at scoped now (that might be what the dbContext is by default anyway, so it could work).

Reply:
Scoped works, going with that & it shouldn't cause any weirdness because it's the default for PuzzleServerContext.